libs/datamodel/connectors/mongodb-datamodel-connector/src/validations.rs

```diff
@@ -160,3 +160,32 @@ pub(crate) fn index_is_not_defined_multiple_times_to_same_fields(index: IndexWal
         *attr.span(),
     ))
 }
+
+/// A field cannot have `@id` and `@unique` attributes at the same time.
+pub(crate) fn unique_cannot_be_defined_to_id_field(index: IndexWalker<'_>, errors: &mut Diagnostics) {
+    let attr = if let Some(attribute) = index.ast_attribute() {
+        attribute
+    } else {
+        return;
+    };
+
+    if !index.is_defined_on_field() {
+        return;
+    }
+
+    if !index.is_unique() {
+        return;
+    }
+
+    let field = index.fields().next().unwrap();
+
+    if !field.is_single_pk() {
+        return;
+    }
+
+    errors.push_error(DatamodelError::new_attribute_validation_error(
+        "An id field cannot hold an additional unique constraint with the current connector.",
```

Comment:
nitpick: I'd have phrased that "The same field cannot be `@id` and `@unique` on MongoDB" (more specific), but this works :+1: